### PR TITLE
Add Two Bucket Test Generator

### PR DESCRIPTION
**Pull request by GKotfis**
I've mixed feelings about what I've done but maybe it's ok ;)
1. SetConstructorInputParameters() is this hardcoded properties array is ok?
2. RenderTestMethodBodyAssert. I tried to do this by using some for loops by Expected values (to avoid this hardcoded property) but without any good result.
Please feel free to give some advice in case something needs change.

### CODE REVIEW

File: exercises/two-bucket/TwoBucket.cs [hunk 1]
```diff
@@ -9,18 +9,18 @@ public enum Bucket
 public class TwoBucketResult
 {
     public int Moves { get; set; }
-    public Bucket GoalBucket { get; set; }
-    public int OtherBucketContents { get; set; }
+    public Bucket Goal_bucket { get; set; }
```
Comment (ErikSchierboom) [inline exercises/two-bucket/TwoBucket.cs]: Please rename this to `GoalBucket`, to conform to standard [.NET naming guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions).

File: exercises/two-bucket/TwoBucket.cs [hunk 2]
```diff
@@ -9,18 +9,18 @@ public enum Bucket
...
+    public int Other_bucket { get; set; }
```
Comment (ErikSchierboom) [inline exercises/two-bucket/TwoBucket.cs]: Please rename this to `OtherBucket`, to conform to standard [.NET naming guidelines](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/general-naming-conventions).

File: exercises/two-bucket/TwoBucket.cs [hunk 3]
```diff
@@ -9,18 +9,18 @@ public enum Bucket
...
 }
 
-public class TwoBuckets
+public class TwoBucket
 {
-    public TwoBuckets(int bucketOneSize, int bucketTwoSize, Bucket startBucket)
+    public TwoBucket(int bucket_one, int bucket_two, Bucket start_bucket)
```
Comment (ErikSchierboom) [inline exercises/two-bucket/TwoBucket.cs]: Could you rename the parameters to `bucketOne`, `bucketTwo` and `startBucket`?

File: generators/Exercises/TwoBucket.cs [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+namespace Generators.Exercises
+{
+    public class TwoBucket : Exercise
+    {
+        protected override void UpdateCanonicalData(CanonicalData canonicalData)
+        {
+            foreach (var canonicalDataCase in canonicalData.Cases)
+            {
+                canonicalDataCase.TestedMethodType = TestedMethodType.Instance;
+                canonicalDataCase.SetConstructorInputParameters(new [] { "bucket_one", "bucket_two", "start_bucket"});
```
Comment (ErikSchierboom) [inline generators/Exercises/TwoBucket.cs]: The `SetConstructorInputParameters` take a `param string[]`, which means you can simplify this to: `SetConstructorInputParameters("bucket_one", "bucket_two", "start_bucket")`

**Comment by GKotfis** [comment]
I was aware of breaking .net coding guidelines here. Wanted match canonical data properties with variables names. But I'll simply replace them in update canonical data method. Stay tuned 😉

**Comment by ErikSchierboom** [comment]
Great! Let me say once again that I'm impressed with the quality of your PR!

**Comment by GKotfis** [comment]
Thanks for the compliment! I think I'll grab another issue :)

**Comment by jpreese** [comment]
Yeah, this looks great @GKotfis! Just as a heads up, since you changed the name of the class, you'll also need to update the `Example.cs` file so that it matches.

File: exercises/two-bucket/TwoBucket.cs [hunk 5]
```diff
@@ -10,17 +10,17 @@ public class TwoBucketResult
 {
     public int Moves { get; set; }
     public Bucket GoalBucket { get; set; }
-    public int OtherBucketContents { get; set; }
+    public int OtherBucket { get; set; }
 }
 
-public class TwoBuckets
+public class TwoBucket
```
Comment (jpreese) [inline exercises/two-bucket/TwoBucket.cs]: Example.cs should be changed as well.

**Comment by ErikSchierboom** [comment]
Everything's green, so I'll merge it. Thanks a lot! 🎉

**Pull request merged**